src/GitHub.Api/Git/Repository.cs

```diff
@@ -64,64 +58,69 @@ private void CacheContainer_OnCacheInvalidated(CacheType cacheType)
 
         private void CacheContainer_OnCacheUpdated(CacheType cacheType, DateTimeOffset offset)
         {
+            var cacheUpdateEvent = new CacheUpdateEvent { UpdatedTimeString = offset.ToString() };
             switch (cacheType)
             {
                 case CacheType.BranchCache:
-                    FireBranchCacheUpdated(offset);
+                    HandleBranchCacheUpdatedEvent(cacheUpdateEvent);
                     break;
 
                 case CacheType.GitLogCache:
-                    FireGitLogCacheUpdated(offset);
+                    HandleGitLogCacheUpdatedEvent(cacheUpdateEvent);
                     break;
 
                 case CacheType.GitStatusCache:
-                    FireGitStatusCacheUpdated(offset);
+                    HandleGitStatucCacheUpdatedEvent(cacheUpdateEvent);
                     break;
 
                 case CacheType.GitLocksCache:
-                    FireGitLocksCacheUpdated(offset);
+                    HandleGitLocksCacheUpdatedEvent(cacheUpdateEvent);
                     break;
 
                 case CacheType.GitUserCache:
                     break;
 
                 case CacheType.RepositoryInfoCache:
-                    FireRepositoryInfoCacheUpdated(offset);
+                    HandleRepositoryInfoCacheUpdatedEvent(cacheUpdateEvent);
                     break;
 
                 default:
                     throw new ArgumentOutOfRangeException(nameof(cacheType), cacheType, null);
             }
         }
 
-        private void FireGitLogCacheUpdated(DateTimeOffset dateTimeOffset)
+        private void HandleRepositoryInfoCacheUpdatedEvent(CacheUpdateEvent cacheUpdateEvent)
         {
-            Logger.Trace("GitLogCacheUpdated {0}", dateTimeOffset);
-            GitLogCacheUpdated?.Invoke(new CacheUpdateEvent { UpdatedTimeString = dateTimeOffset.ToString() });
+            Logger.Trace("RepositoryInfoCache Updated {0}", cacheUpdateEvent.UpdatedTimeString);
+            CurrentBranchChanged?.Invoke(cacheUpdateEvent);
+            CurrentRemoteChanged?.Invoke(cacheUpdateEvent);
+            CurrentBranchAndRemoteChanged?.Invoke(cacheUpdateEvent);
```

Comment:
Do we need an extra event here, given we already raise both events right before?